### ISSUE TITLE
Incorrect class name for public static nested classes

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#5750]
**Affects PMD Version:** 

7.13.0

**Description:**

Given:

```
package com.github.cowwoc.docker;

import java.net.InetAddress;
import java.net.UnknownHostException;

/**
 * Stub.
 */
public final class Testcase
{
	/**
	 * Stub.
	 */
	public static final class NestedClass
	{
		/**
		 * Stub.
		 *
		 * @throws UnknownHostException stub
		 */
		public NestedClass() throws UnknownHostException
		{
			InetAddress.getByName("0.0.0.0");
		}
	}
}
```

I get this warning:

```
[WARNING] PMD Failure: com.github.cowwoc.docker.NestedClass:23 Rule:AvoidUsingHardCodedIP Priority:3 Do not hard code the IP address ${variableName}.
```

Notice that the outer class name is missing.

**Running PMD through:** *Maven*

**Comment by cowwoc** [comment]
Also notice that `${variableName}` is not being replaced by a value in the warning message.

**Comment by oowekyala** [comment]
Related to #4904

**Comment by adangel** [comment]
Duplicate of #4904
